### DISCUSSION TITLE
Generic type on AddMap

### DISCUSSION THREAD

**Discussion by sjblack** [alirezanet/Gridify#159] [Ideas]
Whether this is possible on not I do not know. I thought it wold be nice to have a mapping method that took a generic type for the entity property, something like.

public IQueryBuilder<T> AddMap <TProperty> (string from, Expression<Func<T, TProperty?>> to, Func<string, TProperty>? convertor = null, bool overwrite = true);

**Comment by alirezanet** [comment]
Hi @sjblack,
I'm not sure I see the value in this since Mapper's generic type tells us the property types. Can you please explain the problem you're aiming to solve?"

**Reply by sjblack** [reply]
Hello,thank you for replying. I was thinking on the side of validating the converter returns the type of the entity property. I have put in my own validation thought it would be nice if it wasn't possible for the converter to say return an int when the entity property is a date time for instance.
